src/main/java/com/aws/greengrass/componentmanager/ClientConfiguration.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.componentmanager;
+
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.EncryptionUtils;
+import com.aws.greengrass.util.IotSdkClientFactory;
+import com.aws.greengrass.util.ProxyUtils;
+import com.aws.greengrass.util.RegionUtils;
+import com.aws.greengrass.util.Utils;
+import com.aws.greengrass.util.exceptions.InvalidEnvironmentStageException;
+import com.aws.greengrass.util.exceptions.TLSAuthException;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+
+public class ClientConfiguration {
+    private static final Logger logger = LogManager.getLogger(ClientConfiguration.class);
+
+    /**
+     * Get the greengrass service endpoint.
+     *
+     * @param deviceConfiguration    {@link DeviceConfiguration}
+     * @return service end point
+     */
+    public String getGreengrassServiceEndpoint(DeviceConfiguration deviceConfiguration) {
```

Comment:
All these methods can just be static, this can be a simple utility class.

Reply:
Done